psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c

```diff
@@ -0,0 +1,273 @@
+    
+    // Now the data is decoded
+    g_decode_flag = 1;
+    // From here we will have the acctual values. We can call functions from
+    // here or send it out depending on what we decide to go with 
+}    
+
+// Name: Delay_Timer_ISR_Handler
+// Summary:
+//    - Interrupt Service Routine. Check the Delay_Timer status
+//      and clear the interrupt
+//    - change the Count and Executed status
+
+_CY_ISR(Delay_Timer_ISR_Handler){
+    
+    if(Delay_Timer_ReadPeriod() == DELAY_ACQUISITION) {
+        g_count = 2;                  
+        g_executed=0;             
+    }else if(Delay_Timer_ReadPeriod() == DELAY_PROCESSING) { 
+        g_decode_flag = 1;
+    }
+    
+    // Write 1 to the Contro_Reg_Delay_Timer to disable the Timer function 
+    Control_Reg_Delay_Timer_Write(1);
+    // Read the status register of Delay_Timer to clear interrupt
+    Delay_Timer_ReadStatusRegister();
+    // Clear the Delay_Timer_ISR interrupt
+    Delay_Timer_ISR_ClearPending();
+}    
+
+_CY_ISR(DHT_Pin_ISR_Handler){
+    // Clear the interrupt request of the ISR DHT_Pin_ISR 
+    DHT_Pin_ISR_ClearPending();
+    // Clear the interrupt request of the pin DHT_Pin
+    DHT_Pin_ClearInterrupt();
+    // Read the counter values of Data_Timer for processing
+    g_countvalues_datatimer[g_risingedge_counts]=Data_Timer_ReadCounter();
+    g_risingedge_counts++;
+}
+
+// What dht_init does:
+//    - Initilizes all variables that need to be zeroed out and
+//      otherwise. 
+//    - Writes 1 to the pin as a starting condition
+//    - Start various functions neccessary to read the conditions
+//
+// Paramers: None
+// Returns: None
+
+void dht_init(){
+    g_check = 0;
+    g_decode_flag = 0;
+    g_risingedge_counts = 0;
+    
+    // Starts the data timer and delay timer components
+    Data_Timer_Start();
+    Delay_Timer_Start();
+    
+    // Write Controlregdatatimer and controlregdealytimer value
+    // to 1 to keep them reset
+    Control_Reg_Data_Timer_Write(1);
+    Control_Reg_Delay_Timer_Write(1);
+    
+    // The pin value will be written as 1 as the starting condition
+    DHT_Pin_Write(1);
+    
+    // Initilize dhtpinisr and delaytimeriser
+    DHT_Pin_ISR_StartEx(DHT_Pin_ISR_Handler);
+    Delay_Timer_ISR_StartEx(Delay_Timer_ISR_Handler);
+    
+    // Disable the ISR compondents
+    DHT_Pin_ISR_Disable();
+    Delay_Timer_ISR_Disable();
+    
+    // Initilize g_count to 0 and g_executed to 1
+    g_count = 0;
+    g_executed = 1;
+    
+    g_has_been_executed = 0;
+}   
+
+
+// What dht_start does:
+//    Starts the dht components
+//    Get output from dht11
+//    Send out results
+//
+// Paramers: None
+// Returns: None
+
+
+void dht_start(){
+    
+    // Enable Global Interrupts
+    CyGlobalIntEnable;
+    
+    // This is going to run until decode dht data is called
+    // After this is done, the loop will end, and the data will
+    // be in the respective variables
+    
+    /*
+    switch(g_has_been_executed){
```

Comment:
Everything below here looks like the old blocking implementation. Did you mean to get rid of it?